java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -354,16 +319,10 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_hash(JNIEnv *env, jobje
   JNI_NULL_CHECK(env, column_handles, "array of column handles is null", 0);
 
   try {
-    cudf::jni::native_jpointerArray<cudf::column_view> n_cudf_columns(env, column_handles);
-    std::vector<cudf::column_view> column_views;
-    std::transform(n_cudf_columns.data(), n_cudf_columns.data() + n_cudf_columns.size(),
-                   std::back_inserter(column_views),
-                   [](auto const &p_column) { return *p_column; });
-    cudf::table_view *input_table = new cudf::table_view(column_views);
```

Comment:
I'm not a fan of giant, refactoring PRs that do more than just their main refactor, and this is doing just that.  This is tying an important memory leak bugfix to a completely unrelated change, and it would be much more expedient to review and easier to manage going forward if this fix were committed separately.

Reply:
+1

Reply:
> This is tying an important memory leak bugfix

Since we hadn't noticed it before, I didn't think the memory leak was important, per se. But I do see your point. Moving those fixes to a different commit would make them a candidate for 22.02.

I will update here with a separate PR for just the bug portion of this change.

Reply:
> I will update here with a separate PR...

I'm done reviewing `ColumnViewJni.cpp` and `ColumnVectorJni.cpp. 
The bug PR will have to wait until I'm done scanning through `TableJni.cpp` as well. It shouldn't be long...

Reply:
I have filed #10029 for the memory leaks. I will rebase this branch once that PR is checked in.